src/sentry/middleware/auth.py

```diff
@@ -83,6 +93,20 @@ def process_request(self, request: Request):
             else:
                 # default to anonymous user and use IP ratelimit
                 request.user = SimpleLazyObject(lambda: get_user(request))
+        elif (
+            auth
+            and auth[0].lower() == OrgAuthTokenAuthentication.token_name
+            and force_text(auth[1]).startswith(SENTRY_JWT_PREFIX)
+        ):
```

Comment:
Should we be using a loop to try each of the token types? If `TokenAuthentication` and `OrgAuthTokenAuthentication` had methods to check if the `Authorization` header contained their kind of token you could have logic like

```python
elif auth:
    for authenticator in [TokenAuthentication, OrgAuthTokenAuthentication, ApiKeyAuthentication]:
        if not authentication.accepts_token(auth):
            continue
        try:
            result = authenticator().authenticate(request)
        except AuthenticationFailed:
            result = None
        if result:
```

Reply:
Hmm, there is actually already some code in there to kind of do these checks, I can refactor it a bit to be more reusable!

Reply:
So I implemented `accepts_auth` on the authentication implementations, so we can re-use this properly!